### PR TITLE
Upgrade terraform-provider-google-beta to v5.14.0

### LINKED ISSUE #1604: Upgrade terraform-provider-google-beta to v5.14.0
Release details: https://github.com/hashicorp/terraform-provider-google-beta/releases/tag/v5.14.0

### PR DESCRIPTION
**Pull request by iwahbe**
This PR was generated via `$ upgrade-provider pulumi/pulumi-gcp`.

---

- Upgrading terraform-provider-google-beta from 5.13.0  to 5.14.0.
	Fixes #1604

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -63,28 +63,29 @@ const (
...
 	gcpDataLoss                 = "DataLoss"                 // DataLoss resources
 	gcpDataPlex                 = "DataPlex"                 // DataPlex
 	gcpDataProc                 = "Dataproc"                 // DataProc resources
+	gcpDatabaseMigrationService = "DatabaseMigrationService" // Database Migration Service resources
+	gcpDataform                 = "Dataform"                 // Dataform resources
 	gcpDatastore                = "Datastore"                // Datastore resources
 	gcpDatastream               = "Datastream"               // Datastream resources
 	gcpDeploymentManager        = "DeploymentManager"        // DeploymentManager resources
 	gcpDiagflow                 = "Diagflow"                 // Diagflow resources
+	gcpDiscoveryEngine          = "DiscoveryEngine"          // Discovery Engine
```
Comment (iwahbe) [inline provider/resources.go:88]: Introduced in this PR.

File: provider/resources.go [hunk 2]
```diff
@@ -133,6 +134,7 @@ const (
 	gcpSecretManager            = "SecretManager"            // Secret Manager resources
 	gcpSecureSourceManager      = "SecureSourceManager"      // Secure Source Manager
 	gcpSecurityCenter           = "SecurityCenter"           // Security Center
+	gcpSecurityPosture          = "SecurityPosture"          // Security Posture
```
Comment (iwahbe) [inline provider/resources.go:137]: Introduced in this PR.

**Review by iwahbe — COMMENTED** [review]
The manual part of this PR is limited to introducing 2 new module mappings and applying an alphabetic sort on the mappings tables.

**Review by VenelinMartinov — APPROVED** [review]

File: sdk/dotnet/Firestore/Document.cs [hunk 3]
```diff
@@ -31,62 +31,6 @@ namespace Pulumi.Gcp.Firestore
     /// is allowed for the database parameter.
     /// 
     /// ## Example Usage
-    /// ### Firestore Document Basic
```
Comment (VenelinMartinov) [inline sdk/dotnet/Firestore/Document.cs:34]: have we lost some examples?

**Review by VenelinMartinov — CHANGES_REQUESTED** [review]

**Review by VenelinMartinov — APPROVED** [review]

**Comment by VenelinMartinov** [comment]
Thanks for taking care of it

**Pull request merged**